Sources/Kommunicate/Classes/Controllers/KMConversationController+MessageCharacterLimit.swift

```diff
@@ -20,14 +20,22 @@ extension KMConversationViewController: MessageCharacterLimitDelegate {
     func characterLimit(manager _: MessageCharacterLimitManager, reachedTheLimit _: Int, textCount: Int) {
         let messageLimit = CharacterLimit.charlimit
         let botLimit = CharacterLimit.botCharLimit
-        if isConversationAssignedToDialogflowBot, textCount >= botLimit.soft, textCount <= messageLimit.soft {
-            botCharLimitManager.messageToShow = characterLimitMessage(textCount: textCount, limit: botLimit, isMessageforBot: true)
-            botCharLimitManager.showLimitView(true, disableButton: textCount > botLimit.hard)
-        } else if textCount >= messageLimit.soft {
-            messageCharLimitManager.messageToShow = characterLimitMessage(textCount: textCount, limit: messageLimit, isMessageforBot: false)
-            messageCharLimitManager.showLimitView(true, disableButton: textCount > messageLimit.hard)
+        let cxBotLimit = CharacterLimit.cxBotCharLimit
+        if isConversationAssignedToDialogflowBot && isConversationAssignedToDialogflowCXBot {
+            handleCharacterLimit(textCount: textCount, limit: cxBotLimit, manager: botCharLimitManager, isBotMessage: true)
```

Comment:
All if-else calls the same method with different arguments. We can move these inside handleCharacterLimit for better code readability